internal/daemon/controller/handlers/scopes/scope_service.go

```diff
@@ -355,6 +367,52 @@ func (s Service) DeleteScope(ctx context.Context, req *pbs.DeleteScopeRequest) (
 	return nil, nil
 }
 
+// ListKeys implements the interface pbs.ScopeServiceServer.
+func (s Service) ListKeys(ctx context.Context, req *pbs.ListKeysRequest) (*pbs.ListKeysResponse, error) {
+	if req.GetId() == "" {
+		req.Id = scope.Global.String()
+	}
+	if err := validateListKeysRequest(req); err != nil {
+		return nil, err
+	}
+	authResults := s.authResult(ctx, req.GetId(), action.ListScopeKeys)
+	if authResults.Error != nil {
+		return nil, authResults.Error
+	}
+
+	items, err := s.kmsRepo.ListKeys(ctx, req.GetId())
+	if err != nil {
+		if errors.Is(err, dbw.ErrRecordNotFound) {
```

Comment:
Question; is this a layer violation? Since this package isn't imported already in this file I wasn't sure, but I'm also not sure how to otherwise identify this error.

Reply:
Great, exactly what I was looking for